icu4c/source/common/cmemory.h

```diff
@@ -771,14 +800,29 @@ class MaybeStackVector : protected MemoryPool<T, stackCapacity> {
         return this->create(args...);
     }
 
+    /**
+     * Takes ownership of ptr, adding it to the back of the vector.
+     *
+     * @param ptr Pointer to take ownership of.
+     * @param status Is set to U_MEMORY_ALLOCATION_ERROR if allocation of the
+     * needed pool capacity fails.
+     */
+    void adoptBack(T *ptr, UErrorCode &status) {
```

Comment:
This method has the same error handling issues as `MemoryPool::adopt()` so it raises the same questions.

Reply:
(Let's have this discussion on the MemoryPool::adopt() thread?)